### PR TITLE
[docs] Updating links to guides

### PR DESCRIPTION
**Pull request by kaitlincarter-hc**
After migrating the guides to Learn, we needed to update the links within the docs.

### CODE REVIEW

File: website/source/api/acl-legacy.html.md [hunk 1]
```diff
@@ -12,7 +12,8 @@ the new ACL [Token](/docs/api/acl-token.html) and [Policy](/docs/api/acl-policy.
 
 # ACL HTTP API
 
-These `/acl` endpoints create, update, destroy, and query ACL tokens in Consul. For more information about ACLs, please see the [ACL Guide](/docs/guides/acl.html).
+These `/acl` endpoints create, update, destroy, and query ACL tokens in Consul. For more information about ACLs, please see the 
+[ACL Guide](https://learn.hashicorp.com/consul/security-networking/production-acls).
```
Comment (rboyer) [inline website/source/api/acl-legacy.html.md]: I wonder if there is a `middleman` way to have a list of general named links stored somewhere that can be referred to from other docs to avoid having to inline them all over the place like this.

File: website/source/api/acl-legacy.html.md [hunk 2]
```diff
@@ -347,8 +349,8 @@ This endpoint returns the status of the ACL replication process in the
 datacenter. This is intended to be used by operators, or by automation checking
 the health of ACL replication.
 
-Please see the [ACL Guide](/docs/guides/acl.html#replication) replication
-section for more details.
+Please see the agent configuration [documentation](/docs/agent/options.html#acl_enable_token_replication) for replication
```
Comment (rboyer) [inline website/source/api/acl-legacy.html.md]: This is the wrong link. You can have replication without token replication (just replicating policies and roles).

One of these is likely more suitable. They are both for configuring the token used for _doing_ replication:

* `/docs/agent/options.html#acl_tokens_replication`
* `/docs/agent/options.html#enable_acl_replication`

File: website/source/api/acl/acl.html.md [hunk 3]
```diff
@@ -80,15 +80,15 @@ consider the cluster in a potentially compromised state.
...
+[ACL Guide](https://learn.hashicorp.com/consul/security-networking/production-acls) for more details.
 
 ## Check ACL Replication
 
 This endpoint returns the status of the ACL replication processes in the
 datacenter. This is intended to be used by operators or by automation checking 
 to discover the health of ACL replication.
 
-Please see the [ACL Guide](/docs/guides/acl.html#replication) replication
+Please see the [ACL Guide](/docs/agent/options.html#acl_enable_token_replication) replication
```
Comment (rboyer) [inline website/source/api/acl/acl.html.md]: wrong link (see above)

File: website/source/docs/acl/acl-legacy.html.md [hunk 4]
```diff
@@ -6,7 +6,7 @@ description: |-
   Consul provides an optional Access Control List (ACL) system which can be used to control access to data and APIs. The ACL system is a Capability-based system that relies on tokens which can have fine grained rules applied to them. It is very similar to AWS IAM in many ways.
 ---
 
--> **1.3.0 and earlier:** This guide only applies in Consul versions 1.3.0 and before. If you are using the 1.4.0 or later please use the updated guide [here](/docs/acl/acl-system.html)
+-> **1.3.0 and earlier:** This document only applies in Consul versions 1.3.0 and before. If you are using the 1.4.0 or later please use the updated documentation [here](/docs/acl/acl-system.html)
```
Comment (rboyer) [inline website/source/docs/acl/acl-legacy.html.md]:
```suggestion
-> **1.3.0 and earlier:** This document only applies in Consul versions 1.3.0 and before. If you are using version 1.4.0 or later please use the updated documentation [here](/docs/acl/acl-system.html)
```

File: website/source/api/acl-legacy.html.md (hunk 1 above)
Comment (alvin-huang) [reply website/source/api/acl-legacy.html.md]: pinging @jescalan for any ideas ^

File: website/source/docs/agent/kv.html.md [hunk 5]
```diff
@@ -14,7 +14,7 @@ The Consul KV datastore is located on the servers, but can be accessed by any ag
...
+The KV store can be accessed by the [consul kv CLI subcommands](/docs/commands/kv.html), [HTTP API](/api/kv.html), and Consul UI. To restrict access, enable and configure [ACLs](https://learn.hashicorp.com/consul/security-networking/production-acls). Once the ACL system has been bootstrapped, users and services, will need a valid token with KV [privileges](/docs/agent/acl-rules.html#key-value-rules) to access the the data store, this includes even reads.  We recommend creating a token with limited privileges, for example, you could create a token with write privileges on one key for developers to update the value related to their application.
```
Comment (rboyer) [inline website/source/docs/agent/kv.html.md]: This could benefit from a word wrap while you're here so that future PRs are easier to skim.

File: website/source/docs/agent/options.html.md [hunk 6]
```diff
@@ -684,7 +679,8 @@ default will automatically work with some tooling.
 *   <a name="acl_replication_token_legacy"></a><a href="#acl_replication_token_legacy">`acl_replication_token`</a> -
     **Deprecated in Consul 1.4.0. See the [`acl.tokens.replication`](#acl_tokens_replication) field instead.**
     Only used for servers outside the [`primary_datacenter`](#primary_datacenter) running Consul 0.7 or later.
-    When provided, this will enable [ACL replication](/docs/guides/acl.html#replication) using this
+    When provided, this will enable [
+    ACL replication using this
```
Comment (rboyer) [inline website/source/docs/agent/options.html.md]: missing link

File: website/source/docs/enterprise/connect-multi-datacenter/index.html.md [hunk 7]
```diff
@@ -14,7 +14,7 @@ and federation of Certificate Authority trust.
 
 Sidecar proxy's [upstream configuration](/docs/connect/proxies.html#upstream-configuration-reference) 
 may specify an alternative datacenter or a prepared query that can address services 
-in multiple datacenters (such as the [geo failover](/docs/guides/geo-failover.html) pattern).
+in multiple datacenters (such as the [geo failover](https://learn.hashicorp.com/consul/developer-discovery/geo-failoverl) pattern).
```
Comment (rboyer) [inline website/source/docs/enterprise/connect-multi-datacenter/index.html.md]: should this link end with a stray `l`?

File: website/source/docs/upgrade-specific.html.md [hunk 8]
```diff
@@ -167,7 +167,7 @@ before proceeding.
...
+Consul 1.0 (and earlier versions of Consul when running with [Raft protocol 3](/docs/agent/options.html#_raft_protocol) had an issue where performing rolling updates of Consul servers could result in an outage from old servers remaining in the cluster. [Autopilot](https://learn.hashicorp.com/consul/day-2-operations/autopilot) would normally remove old servers when new ones come online, but it was also waiting to promote servers to voters in pairs to maintain an odd quorum size. The pairwise promotion feature was removed so that servers become voters as soon as they are stable, allowing Autopilot to remove old servers in a safer way.
```
Comment (rboyer) [inline website/source/docs/upgrade-specific.html.md]: this could benefit from a proactive drive-by wordwrapping

File: website/source/docs/upgrade-specific.html.md [hunk 9]
```diff
@@ -177,15 +177,15 @@ Consul 1.0 has several important breaking changes that are documented here. Plea
...
+Raft protocol version 3 requires Consul running 0.8.0 or newer on all servers in order to work, so if you are upgrading with older servers in a cluster then you will need to set this back to 2 in order to upgrade. See [Raft Protocol Version Compatibility](/docs/upgrade-specific.html#raft-protocol-version-compatibility) for more details. Also the format of `peers.json` used for outage recovery is different when running with the latest Raft protocol. See [Manual Recovery Using peers.json](https://learn.hashicorp.com/consul/day-2-operations/outage#manual-recovery-using-peers-json) for a description of the required format.
```
Comment (rboyer) [inline website/source/docs/upgrade-specific.html.md]: this could benefit from a proactive drive-by wordwrapping

File: website/source/docs/upgrade-specific.html.md (hunk 8 above)
Comment (rboyer) [reply website/source/docs/upgrade-specific.html.md]: most of the larger changes in this file could benefit from a wordwrap too while you're in here

File: website/source/docs/enterprise/connect-multi-datacenter/index.html.md (hunk 7 above)
Comment (kaitlincarter-hc) [reply website/source/docs/enterprise/connect-multi-datacenter/index.html.md]: Nope.

Comment (kaitlincarter-hc) [inline website/source/docs/enterprise/connect-multi-datacenter/index.html.md]:
```suggestion
in multiple datacenters (such as the [geo failover](https://learn.hashicorp.com/consul/developer-discovery/geo-failover) pattern).
```

File: website/source/api/acl-legacy.html.md (hunk 2 above)
Comment (kaitlincarter-hc) [inline website/source/api/acl-legacy.html.md]:
```suggestion
Please see the [ACL Replication Guide](https://learn.hashicorp.com/consul/day-2-operations/acl-replication) for more details. 
```

File: website/source/api/acl/acl.html.md (hunk 3 above)
Comment (kaitlincarter-hc) [inline website/source/api/acl/acl.html.md]:
```suggestion
Please see the [ACL Replication Guide](https://learn.hashicorp.com/consul/day-2-operations/acl-replication)
```

File: website/source/docs/agent/options.html.md [hunk 10]
```diff
@@ -684,7 +679,8 @@ default will automatically work with some tooling.
 *   <a name="acl_replication_token_legacy"></a><a href="#acl_replication_token_legacy">`acl_replication_token`</a> -
     **Deprecated in Consul 1.4.0. See the [`acl.tokens.replication`](#acl_tokens_replication) field instead.**
     Only used for servers outside the [`primary_datacenter`](#primary_datacenter) running Consul 0.7 or later.
-    When provided, this will enable [ACL replication](/docs/guides/acl.html#replication) using this
+    When provided, this will enable [
```
Comment (kaitlincarter-hc) [inline website/source/docs/agent/options.html.md]:
```suggestion
    When provided, this will enable [ACL replication](https://learn.hashicorp.com/consul/day-2-operations/acl-replication) using this
```

Comment (kaitlincarter-hc) [inline website/source/docs/agent/options.html.md]:
```suggestion
```

File: website/source/docs/faq.html.md [hunk 11]
```diff
@@ -103,5 +103,5 @@ an RPC request to the remote Consul servers for that resource and return the res
 If the remote datacenter is not available, then those resources will also not be
 available, but that won't otherwise affect the local datacenter. There are some special
 situations where a limited subset of data can be replicated, such as with Consul's built-in
-[ACL replication](/docs/guides/acl.html#outages-and-acl-replication) capability, or
+ACL replication capability, or
```
Comment (kaitlincarter-hc) [inline website/source/docs/faq.html.md]:
```suggestion
[ACL replication](https://learn.hashicorp.com/consul/day-2-operations/acl-replication) capability, or
```

File: website/source/docs/internals/architecture.html.md [hunk 12]
```diff
@@ -120,7 +120,7 @@ an RPC request to the remote Consul servers for that resource and return the res
 If the remote datacenter is not available, then those resources will also not be
 available, but that won't otherwise affect the local datacenter. There are some special
 situations where a limited subset of data can be replicated, such as with Consul's built-in
-[ACL replication](/docs/guides/acl.html#outages-and-acl-replication) capability, or
+ACL replication capability, or
```
Comment (kaitlincarter-hc) [inline website/source/docs/internals/architecture.html.md]:
```suggestion
[ACL replication](https://learn.hashicorp.com/consul/day-2-operations/acl-replication) capability, or
```

File: website/source/api/acl-legacy.html.md (hunk 1 above)
Comment (kaitlincarter-hc) [reply website/source/api/acl-legacy.html.md]: The downside of storing them separately is the possible confusing of figuring out where a link goes. It could add another step when editing docs too, editors would need to go to another file to track down links.

File: website/source/api/acl-legacy.html.md (hunk 2 above)
Comment (kaitlincarter-hc) [reply website/source/api/acl-legacy.html.md]: Updated to the new ACL Replication guide.

File: website/source/docs/agent/kv.html.md (hunk 5 above)
Comment (kaitlincarter-hc) [reply website/source/docs/agent/kv.html.md]: I updated the entire doc.

File: website/source/docs/upgrade-specific.html.md (hunk 8 above)
Comment (kaitlincarter-hc) [reply website/source/docs/upgrade-specific.html.md]: Done.

File: website/source/docs/agent/encryption.html.md [hunk 13]
```diff
@@ -66,7 +66,7 @@ Consul supports using TLS to verify the authenticity of servers and clients. To
 Consul requires that all clients and servers have key pairs that are generated by a single
 Certificate Authority. This can be a private CA, used only internally. The
 CA then signs keys for each of the agents, as in
-[this tutorial on generating both a CA and signing keys](/docs/guides/creating-certificates.html). 
+[this tutorial on generating both a CA and signing keys](/https://learn.hashicorp.com/consul/security-networking/certificates).
```
Comment (alvin-huang) [inline website/source/docs/agent/encryption.html.md]: The slash in the front needs to be removed. This is causing a broken link :)

Comment (kaitlincarter-hc) [inline website/source/docs/agent/encryption.html.md]:
```suggestion
[this tutorial on generating both a CA and signing keys](https://learn.hashicorp.com/consul/security-networking/certificates). 
```
